src/Moderation/DomainPatternBlocklist.php

```diff
@@ -0,0 +1,226 @@
+<?php
+/**
+ * @copyright Copyright (C) 2010-2022, the Friendica project
+ *
+ * @license GNU AGPL version 3 or any later version
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ */
+
+namespace Friendica\Moderation;
+
+use Exception;
+use Friendica\App\BaseURL;
+use Friendica\Core\Config\Capability\IManageConfigValues;
+use Friendica\Core\L10n;
+use Friendica\Database\Database;
+use Friendica\Util\Emailer;
+
+class DomainPatternBlocklist
+{
+	const DEFAULT_REASON = 'blocked';
+
+	/** @var IManageConfigValues */
+	private $config;
+
+	/** @var Database */
+	private $db;
+
+	/** @var Emailer */
+	private $emailer;
+
+	/** @var L10n */
+	private $l10n;
+
+	/** @var BaseURL */
+	private $baseUrl;
+
+	public function __construct(IManageConfigValues $config, Database $db, Emailer $emailer, L10n $l10n, BaseURL $baseUrl)
+	{
+		$this->config  = $config;
+		$this->db      = $db;
+		$this->emailer = $emailer;
+		$this->l10n    = $l10n;
+		$this->baseUrl = $baseUrl;
+	}
+
+	public function get(): array
+	{
+		return $this->config->get('system', 'blocklist', []);
+	}
+
+	public function set(array $blocklist): bool
+	{
+		$result = $this->config->set('system', 'blocklist', $blocklist);
+		if ($result) {
+			$this->notifyAll();
+		}
+
+		return $result;
+	}
+
+	/**
+	 * @param string      $pattern
+	 * @param string|null $reason
+	 * @return int 0 if the block list couldn't be saved, 1 if the pattern was added, 2 if it was updated in place
+	 */
+	public function addPattern(string $pattern, string $reason = null): int
+	{
+		$update = false;
+
+		$blocklist = [];
+		foreach ($this->get() as $blocked) {
+			if ($blocked['domain'] === $pattern) {
+				$blocklist[] = [
+					'domain' => $pattern,
+					'reason' => $reason ?? self::DEFAULT_REASON,
+				];
+
+				$update = true;
+			} else {
+				$blocklist[] = $blocked;
+			}
+		}
+
+		if (!$update) {
+			$blocklist[] = [
+				'domain' => $pattern,
+				'reason' => $reason ?? self::DEFAULT_REASON,
+			];
+		}
+
+		$this->set($blocklist);
+
+		return $this->set($blocklist) ? ($update ? 2 : 1) : 0;
+	}
+
+	/**
```

Comment:
Can you please describe what this method does?

Reply:
No any better than its name already suggests.

Reply:
Not any better than its name suggests.